### ISSUE TITLE
[Input] Uncontrolled to controlled warning

### ISSUE THREAD

**Issue by ivaylogabrovsky** [telus/tds-core#1158]
## Description
React Warning for @tds/core-input v3.0.3
Warning: A component is changing an uncontrolled input of type text to be controlled.

- Description of the defect, observations
- The result you were expecting

## Reproduction Steps
```
initialState = {
  value: '',
}
const onChange = evt => {
  setState({ value: evt.target.value })
}
;<Box between={2}>
  <Input label="First name" type="text" value={state.value} onChange={onChange} />
</Box>
```
1. Select Input component
2. Click any character
3. Observe React Warning

## Recommendation

Observe the `value` prop and ensure the component remains 'controlled' even if an empty string is passed.

## Meta

- TDS component version: @tds/core-input v3.0.3
- Willing to develop solution: Yes/No
- Has workaround: Yes/No set initial value to one space ' '
- High impact: No
- Also observed: Component does not re-render in case of state change

## Screenshots

```
Warning: A component is changing an uncontrolled input of type text to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
    in input (created by Context.Consumer)
    in StyledComponent (created by Input__StyledInput)
    in Input__StyledInput (created by Input)
    in div (created by Context.Consumer)
    in StyledComponent (created by styled.div)
    in styled.div (created by Input)
    in div (created by Box)
    in Box (created by Input)
    in Input (created by AccountInfoForm)
    in div (created by Box)
    in Box (created by BodyContainer)
    in div (created by Col)
    in Col (created by Subscriber)
    in Subscriber (created by Col)
    in Col (created by BodyContainer)
    in div (created by Row)
    in Row (created by Row)
    in Row (created by BodyContainer)
    in div (created by Grid)
    in Grid (created by FlexGrid)
    in Broadcast (created by FlexGrid)
    in FlexGrid (created by BodyContainer)
    in BodyContainer (created by AccountInfoForm)
    in form (created by AccountInfoForm)
    in AccountInfoForm (created by AccountInfo)
    in div (created by Box)
    in Box (created by PageContainer)
    in div (created by Col)
    in Col (created by Subscriber)
    in Subscriber (created by Col)
    in Col (created by PageContainer)
    in div (created by Row)
    in Row (created by Row)
    in Row (created by PageContainer)
    in div (created by Grid)
    in Grid (created by FlexGrid)
    in Broadcast (created by FlexGrid)
    in FlexGrid (created by PageContainer)
    in div (created by Box)
    in Box (created by PageContainer)
    in PageContainer (created by AccountInfo)
    in div (created by AccountInfo)
    in AccountInfo (created by Session)
    in Session (created by Locale)
    in Locale (created by lifecycle(Locale))
    in lifecycle(Locale) (created by MapImmutableToPlain)
    in MapImmutableToPlain (created by Context.Consumer)
    in Connect(MapImmutableToPlain) (created by AsyncComponent)
    in AsyncComponent (created by RouterContext)
    in div (created by App)
    in App (created by RouterContext)
    in RouterContext (created by Router)
    in Router
    in LocaleProvider
    in Provider
```

**Comment by theetrain** [comment]
Note to developer when patching:

We recommend passing in the `value` prop as-is, and adding a handler for `onKeyDown` when `type="email"`.

```diff
+const handleKeyDown = (e) => {
+  if (type === 'email' && e.key === ' ') {
+    e.preventDefault()
+  }
+  rest.onKeyDown && rest.onKeyDown()
+}

<StyledInput
  {...safeRest(rest)}
  type={type}
  ref={ref}
  id={fieldId.identity()}
- {...(value ? { value } : {})}
+ value={value}
+ onKeyDown={handleKeyDown}
  feedback={feedback}
  aria-invalid={feedback === 'error'}
  aria-describedby={errorId || hintId || helperId || undefined}
  onFocus={handleFocus}
  onBlur={handleBlur}
/>
```

**Comment by ah-arch** [comment]
Our team encountered this earlier and eventually found this issue, but noticed that the code snippet in the documentation may be out of date.

The [readme](https://github.com/telus/tds-core/blob/bca66b495092168f725719f0ede5441acadc7989/packages/Input/Input.md#minimal-usage) has a `defaultValue` prop which no longer exists in the [props](https://github.com/telus/tds-core/blob/bca66b495092168f725719f0ede5441acadc7989/packages/Input/Input.jsx#L206-L257) list:
![defaultValue prop](https://user-images.githubusercontent.com/4450690/60993462-865a7e00-a31c-11e9-8873-49551422c28f.png)

**Comment by theetrain** [comment]
Thanks for the callout, @ah-arch. `defaultValue` is a feature part of [React Uncontrolled Components](https://reactjs.org/docs/uncontrolled-components.html#default-values). Though we added it as an optional prop in the underlying FormField component, we can add it as an explicit prop in all TDS Form components for clearer usage.

**Comment by jraff** [comment]
@ivaylogabrovsky This has been fixed with @tds/core-input@3.0.9.
